pkg/releaseutil/kind_sorter.go

```diff
@@ -138,9 +139,10 @@ func lessByKind(a interface{}, b interface{}, kindA string, kindB string, o Kind
 	second, bok := ordering[kindB]
 
 	if !aok && !bok {
-		// if both are unknown then sort alphabetically by kind, keep original order if same kind
-		if kindA != kindB {
-			return kindA < kindB
+		// if install, keep CRs installing order as the order of CRs in the raw manifests
+		// if uninstall, keep CRs uninstalling order as the reversed order of CRs in the raw manifests
+		if reflect.DeepEqual(o, UninstallOrder) {
+			return true
```

Comment:
This change doesn't work exactly as expected. In the tests change the order of NginxVhost and Registration and they will fail. The sorting isn't really working when both kinds are unknown. You'll need to check if it's installing or uninstalling and do the comparison appropriately in each case.

Reply:
I have updated this PR, please review, thanks.